### ISSUE TITLE
Wiredash crashing during performLayout()

### ISSUE THREAD

**Issue by gradycsjohnson** [wiredashio/wiredash-sdk#123]
First off, I LOVE Wiredash! I had been using it to make tickets for my project, but I guess I upgraded some packages a while back and now Wiredash is crashing the app every time I try to open it in web (Chrome).

Flutter channel: dev

I've already changed my dependency to avoid the window error as per:
https://github.com/wiredashio/wiredash-sdk/pull/109#issue-535187076

Any help would be greatly appreciated! Thanks :)

Here's the stack trace:

```
══╡ EXCEPTION CAUGHT BY RENDERING LIBRARY
╞═════════════════════════════════════════════════════════
The following assertion was thrown during performLayout():
Assertion failed:
org-dartlang-sdk:///flutter_web_sdk/lib/_engine/engine/canvaskit/text.dart:675:14
glyphs.length == codeUnitsSupported.length
is not true

The relevant error-causing widget was:
  Text
  file:///home/gj/snap/flutter/common/flutter/.pub-cache/git/wiredash-sdk-e4d5ab8574600c023c3a4e
  564d07c5cecb86fd5b/lib/src/feedback/feedback_sheet.dart:131:19

When the exception was thrown, this was the stack:
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 236:49  throw_
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 29:3    assertFailed
lib/_engine/engine/canvaskit/text.dart 675:49
[_ensureFontsSupportText]
lib/_engine/engine/canvaskit/text.dart 700:5                                  addText
packages/flutter/src/painting/text_span.dart 204:14                           build
packages/flutter/src/painting/text_painter.dart 569:7                         layout
packages/flutter/src/rendering/paragraph.dart 515:18                          [_layoutText]
packages/flutter/src/rendering/paragraph.dart 538:5
[_layoutTextWithConstraints]
packages/flutter/src/rendering/paragraph.dart 651:5                           performLayout
packages/flutter/src/rendering/object.dart 1784:7                             layout
packages/flutter/src/rendering/layout_helper.dart 54:10                       layoutChild
packages/flutter/src/rendering/flex.dart 830:43                               [_computeSizes]
packages/flutter/src/rendering/flex.dart 932:32                               performLayout
packages/flutter/src/rendering/object.dart 1784:7                             layout
packages/flutter/src/rendering/proxy_box.dart 118:7                           performLayout
packages/flutter/src/rendering/object.dart 1784:7                             layout
packages/flutter/src/rendering/layout_helper.dart 54:10                       layoutChild
packages/flutter/src/rendering/flex.dart 830:43                               [_computeSizes]
packages/flutter/src/rendering/flex.dart 932:32                               performLayout
packages/flutter/src/rendering/object.dart 1784:7                             layout
packages/flutter/src/rendering/shifted_box.dart 226:5                         performLayout
packages/flutter/src/rendering/object.dart 1784:7                             layout
packages/flutter/src/rendering/proxy_box.dart 118:7                           performLayout
packages/flutter/src/rendering/object.dart 1784:7                             layout
packages/flutter/src/rendering/proxy_box.dart 279:7                           performLayout
packages/flutter/src/rendering/object.dart 1784:7                             layout
packages/flutter/src/rendering/layout_helper.dart 54:10                       layoutChild
packages/flutter/src/rendering/flex.dart 830:43                               [_computeSizes]
packages/flutter/src/rendering/flex.dart 932:32                               performLayout
packages/flutter/src/rendering/object.dart 1784:7                             layout
packages/flutter/src/rendering/shifted_box.dart 226:5                         performLayout
packages/flutter/src/rendering/object.dart 1784:7                             layout
packages/flutter/src/rendering/proxy_box.dart 118:7                           performLayout
packages/flutter/src/rendering/object.dart 1784:7                             layout
packages/flutter/src/rendering/proxy_box.dart 118:7                           performLayout
packages/flutter/src/rendering/custom_paint.dart 546:11                       performLayout
packages/flutter/src/rendering/object.dart 1784:7                             layout
packages/flutter/src/rendering/proxy_box.dart 118:7                           performLayout
packages/flutter/src/rendering/proxy_box.dart 1371:11                         performLayout
packages/flutter/src/rendering/object.dart 1784:7                             layout
packages/flutter/src/rendering/shifted_box.dart 430:7                         performLayout
packages/flutter/src/rendering/object.dart 1784:7                             layout
packages/flutter/src/rendering/proxy_box.dart 118:7                           performLayout
packages/flutter/src/rendering/object.dart 1784:7                             layout
packages/flutter/src/rendering/proxy_box.dart 118:7                           performLayout
packages/flutter/src/rendering/object.dart 1784:7                             layout
packages/flutter/src/rendering/proxy_box.dart 118:7                           performLayout
packages/flutter/src/rendering/object.dart 1784:7                             layout
packages/flutter/src/rendering/layout_helper.dart 54:10                       layoutChild
packages/flutter/src/rendering/stack.dart 570:43                              [_computeSize]
packages/flutter/src/rendering/stack.dart 597:12                              performLayout
packages/flutter/src/rendering/object.dart 1784:7                             layout
packages/flutter/src/rendering/shifted_box.dart 226:5                         performLayout
packages/flutter/src/rendering/object.dart 1784:7                             layout
packages/flutter/src/widgets/single_child_scroll_view.dart 558:7              performLayout
packages/flutter/src/rendering/object.dart 1784:7                             layout
packages/flutter/src/rendering/proxy_box.dart 118:7                           performLayout
packages/flutter/src/rendering/object.dart 1784:7                             layout
packages/flutter/src/rendering/proxy_box.dart 118:7                           performLayout
packages/flutter/src/rendering/object.dart 1784:7                             layout
packages/flutter/src/rendering/proxy_box.dart 118:7                           performLayout
packages/flutter/src/rendering/object.dart 1784:7                             layout
packages/flutter/src/rendering/proxy_box.dart 118:7                           performLayout
packages/flutter/src/rendering/object.dart 1784:7                             layout
packages/flutter/src/rendering/proxy_box.dart 118:7                           performLayout
packages/flutter/src/rendering/object.dart 1784:7                             layout
packages/flutter/src/rendering/proxy_box.dart 118:7                           performLayout
packages/flutter/src/rendering/object.dart 1784:7                             layout
packages/flutter/src/rendering/proxy_box.dart 279:7                           performLayout
packages/flutter/src/rendering/object.dart 1784:7                             layout
packages/flutter/src/rendering/shifted_box.dart 430:7                         performLayout
packages/flutter/src/rendering/object.dart 1784:7                             layout
packages/flutter/src/rendering/shifted_box.dart 226:5                         performLayout
packages/flutter/src/rendering/object.dart 1784:7                             layout
packages/flutter/src/rendering/proxy_box.dart 118:7                           performLayout
packages/flutter/src/rendering/object.dart 1784:7                             layout
packages/flutter/src/rendering/proxy_box.dart 118:7                           performLayout
packages/flutter/src/rendering/object.dart 1784:7                             layout
packages/flutter/src/rendering/layout_helper.dart 54:10                       layoutChild
packages/flutter/src/rendering/stack.dart 570:43                              [_computeSize]
packages/flutter/src/rendering/stack.dart 597:12                              performLayout
packages/flutter/src/rendering/object.dart 1784:7                             layout
packages/flutter/src/rendering/proxy_box.dart 118:7                           performLayout
packages/flutter/src/rendering/object.dart 1784:7                             layout
packages/flutter/src/rendering/proxy_box.dart 118:7                           performLayout
packages/flutter/src/rendering/object.dart 1784:7                             layout
packages/flutter/src/rendering/proxy_box.dart 118:7                           performLayout
packages/flutter/src/rendering/object.dart 1784:7                             layout
packages/flutter/src/rendering/proxy_box.dart 3360:14                         performLayout
packages/flutter/src/rendering/object.dart 1784:7                             layout
packages/flutter/src/rendering/proxy_box.dart 118:7                           performLayout
packages/flutter/src/rendering/object.dart 1784:7                             layout
packages/flutter/src/widgets/overlay.dart 743:14                              performLayout
packages/flutter/src/rendering/object.dart 1641:7
[_layoutWithoutResize]
packages/flutter/src/rendering/object.dart 884:17                             flushLayout
packages/flutter/src/rendering/binding.dart 453:19                            drawFrame
packages/flutter/src/widgets/binding.dart 874:13                              drawFrame
packages/flutter/src/rendering/binding.dart 319:5
[_handlePersistentFrameCallback]
packages/flutter/src/scheduler/binding.dart 1144:15
[_invokeFrameCallback]
packages/flutter/src/scheduler/binding.dart 1082:9                            handleDrawFrame
packages/flutter/src/scheduler/binding.dart 998:5                             [_handleDrawFrame]
lib/_engine/engine/platform_dispatcher.dart 896:13                            invoke
lib/_engine/engine/platform_dispatcher.dart 145:5                             invokeOnDrawFrame
lib/_engine/engine.dart 259:45                                                <fn>

The following RenderObject was being processed when the exception was fired:
RenderParagraph#82d65 relayoutBoundary=up30 NEEDS-LAYOUT NEEDS-PAINT:
  creator: RichText ← Text ← Column ← FadeTransition ← AnimatedFadeIn ← AnimatedBuilder ← Column
  ←
    Padding ← DecoratedBox ← ConstrainedBox ← Container ← Column ← ⋯
  parentData: offset=Offset(0.0, 0.0); flex=null; fit=null (can use size)
  constraints: BoxConstraints(0.0<=w<=416.0, 0.0<=h<=Infinity)
  size: MISSING
  textAlign: start
  textDirection: ltr
  softWrap: wrapping at box width
  overflow: clip
  locale: en_US
  maxLines: unlimited
This RenderObject had the following child:
    text: TextSpan
════════════════════════════════════════════════════════════════════════════════════════════════
════
Another exception was thrown: Assertion failed:
file:///home/gj/snap/flutter/common/flutter/packages/flutter/lib/src/rendering/box.dart:1940:12


If I let it run a bit longer I end up with this error:

Error: Cannot hit test a render box with no size.
The hitTest() method was called on this RenderBox:
RenderConstrainedBox#068b9 relayoutBoundary=up5:
  needs compositing
  creator: ConstrainedBox ← Align ← Padding ← Container ←
  _DownGestureDetector<dynamic> ← FractionalTranslation ←
  SlideTransition ← FractionalTranslation ← SlideTransition ←
  DismissablePageTransition ← Stack ← WillPopScope ← ⋯
  parentData: offset=Offset(0.0, 0.0) (can use size)
  constraints: BoxConstraints(0.0<=w<=1848.0, 0.0<=h<=981.0)
  size: MISSING
  additionalConstraints: BoxConstraints(0.0<=w<=480.0,
  0.0<=h<=Infinity)
Although this node is not marked as needing layout, its size is not
set.
A RenderBox object must have an explicit size before it can be
hit-tested. Make sure that the RenderBox in question sets its size
during layout.
    at Object.throw_ [as throw]
    (http://localhost:46359/dart_sdk.js:5336:11)
    at
    http://localhost:46359/packages/flutter/src/rendering/layer.dar
    t.lib.js:4593:21
    at proxy_box.RenderConstrainedBox.new.hitTest
    (http://localhost:46359/packages/flutter/src/rendering/layer.da
    rt.lib.js:4596:26)
```

**Comment by passsy** [comment]
Thanks for reporting the issue. Currently, we only support the flutter `stable` channel but we'll officially support the `beta` and `dev` channels in the next week. The latest breaking changes really make this a necessity.

The crash looks like a emoji bug. Does it directly crash when opening the feedback sheet or only after entering text?

**Comment by gradycsjohnson** [comment]
Thanks for getting back to me so quickly!

Yes it directly crashes when trying to open the feedback sheet.

**Comment by passsy** [comment]
I could reproduce this bug with Flutter `1.26.0-17.1.pre`. It seems to be fixed with Flutter `1.26.0-17.2.pre` printing the following message when showing an emoji:

```
Could not find a set of Noto fonts to display all missing characters. Please add a font asset for the missing characters. See: https://flutter.dev/docs/cookbook/design/fonts
```
